src/Microsoft.ML.OnnxTransform/DnnImageFeaturizerTransform.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.StaticPipe;
+using Microsoft.ML.StaticPipe.Runtime;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.ML.Transforms
+{
+    // This is a helper class that is required to use the DnnImageFeaturizer estimator.
+    // Note that by default, it is not usable as it does not have any valid methods that return an EstimatorChain that
+    // is used by the DnnImageFeaturizeEstimator.
+    // In order to use this, at least one model project with the corresponding extension methods must by included.
+    // See Microsoft.ML.DNNImageFeaturizer.ResNet18 for an example.
+    public sealed class DnnImageModelSelector
+    {
+    }
+
+    // The Dnn Image Featurizer is just a wrapper around two OnnxTransforms with present pretrained DNN models.
+    // Note that because of this, it only works on Windows machines as that is a constraint of the OnnxTransform.
+    public sealed class DnnImageFeaturizerEstimator : IEstimator<TransformerChain<OnnxTransform>>
+    {
+        private readonly EstimatorChain<OnnxTransform> _modelChain;
+
+        // The function passed in to this method is expected to be an extension method on the DnnImageModelSelector class
+        // that creates a chain of two OnnxTransforms with specific models included together with that extension method.
+        // For an example, see Microsoft.ML.DnnImageFeaturizer.ResNet18.
+        public DnnImageFeaturizerEstimator(IHostEnvironment env, Func<DnnImageModelSelector, IHostEnvironment, string, string, EstimatorChain<OnnxTransform>> model, string input, string output)
+        {
+            _modelChain = model(new DnnImageModelSelector(), env, input, output);
+        }
+
+        public TransformerChain<OnnxTransform> Fit(IDataView input)
+        {
+            return _modelChain.Fit(input);
+        }
```

Comment:
Can OnnxTransform fit now? I assume that this module is built from pre-trained models only.

Reply:
OnnxTransform is a trivial estimator. Which means that fit actually just checks the schema but does no real fitting. The reason this is here is to adhere to the standard interface. Note this is also how other non trainable estimators work (can see OnnxScoringEstimator for an example).

---
In reply to: [232401935](https://github.com/dotnet/machinelearning/pull/1447#discussion_r232401935) [](ancestors = 232401935)

Reply:
Can you add this note to this function?

---
In reply to: [232403204](https://github.com/dotnet/machinelearning/pull/1447#discussion_r232403204) [](ancestors = 232403204,232401935)

Reply:
Also, do you mean that "OnnxScoringEstimator" is an estimator? Here is a part of OnnxTransform --- `public sealed class OnnxTransform : ITransformer, ICanSaveModel`


---
In reply to: [232407856](https://github.com/dotnet/machinelearning/pull/1447#discussion_r232407856) [](ancestors = 232407856,232403204,232401935)

Reply:
Done!

---
In reply to: [232407856](https://github.com/dotnet/machinelearning/pull/1447#discussion_r232407856) [](ancestors = 232407856,232403204,232401935)

Reply:
Yes, sorry. The OnnxScoringEstimator which this is based on is a trivial estimator.

---
In reply to: [232408322](https://github.com/dotnet/machinelearning/pull/1447#discussion_r232408322) [](ancestors = 232408322,232407856,232403204,232401935)